### ISSUE TITLE
Flash actionscript variables not being passed using loadVariables

### ISSUE THREAD

**Issue by charco66** [ruffle-rs/ruffle#2584]
I have many flash files with actionscript 2 that send variables on to a php mailer script. These no longer work with ruffle.

function endGame() {
	fillForm();
	summary.sendit.onRelease = function() {
		// send variables in form movieclip (the textfields)
		// to email PHP page which will send the mail
		summary.loadVariables("../mail/mc_test.php","POST");
		summary.onEnterFrame = function() {
			_root.attachMovie("thanks","thanks",_root.getNextHighestDepth(),{_x:0, _y:0});
			removeMovieClip(summary);
		};
		removeMovieClip("formula");
		quickReset();
		my_txt.removeTextField();
	};
}

The php mailer does no receive the variables loaded by the actionscript.

**Comment by charco66** [comment]
Yes, I would like to pass variables from flash (ruffle) to php

**Comment by n0samu** [comment]
@charco66 Can you provide a sample Flash file that passes variables correctly in Flash Player but fails in Ruffle?

**Comment by flashuser2** [comment]
Hi, I am having the same problem sending mails to users.
I am using a form similar to the attached file
[flashmo_102_contact_form.zip](https://github.com/ruffle-rs/ruffle/files/11129354/flashmo_102_contact_form.zip)

**Comment by n0samu** [comment]
Hi, can you explain what problem you're having exactly? I can't test the PHP script, but Ruffle appears to be sending all of the necessary data correctly in the POST request body, so I don't see what could be wrong.

**Comment by flashuser2** [comment]
Hi, I have tested the ruffle-version on this site :  www.myschool.nu/flashmail.html  
When I press "send", a message confirms, that the mail was sent, - but nothing is received at the other end. (please see the attached screenshot.png)

I am not really sure what the problem is.  Both the code from the fla.file and the php-script have worked fine for years - in flashplayer though. 
     

[screenshot.zip](https://github.com/ruffle-rs/ruffle/files/11130216/screenshot.zip)

**Comment by n0samu** [comment]
I just tried filling out the contact form on your site using a Flash-enabled browser. Was the mail received then?
This is the browser I used: https://archive.org/details/basilisk-portable-with-flash

**Comment by charco66** [comment]
Hi Rufflers:

Can I contribute further to this issue:

I have many tests that operate in flash and send the results in the following manner:
The variables are all loaded because they appear in the flash at the end of the test.

This is the actionscript 2.0 function used to send all of the variables to the php mailer:


********************************************************
//  ------------------------  email function   ----------------------------

function endGame() {
    fillForm();
    summary.sendit.onRelease = function() {
        // send variables in form movieclip (the textfields)
        // to email PHP page which will send the mail
        summary.loadVariables("../mail/mc_test.php","POST");
        summary.onEnterFrame = function() {// do or show whatever you want after the form is sent, splash screen?
            _root.attachMovie("thanks","thanks",_root.getNextHighestDepth(),{_x:0, _y:0});
            removeMovieClip(summary);
        };
        removeMovieClip("formula");
        quickReset();
    };
}
********************************************************

The php mailer program accepts the data 

********************************************************

<?php

error_reporting(E_ALL);


$data0=$_POST["testName"];
$data1=$_POST["realname"];
$data2=$_POST["marks"];
$data3=$_POST["asked"];
$data4=$_POST["wrongOnes"];
$data5=$_POST["txtFile"];
$data6=$_POST["destination"];
$data7=$_POST["percentage"];


$sendTo = ***@***.***".", ".$data6;
$subject = $data0;


// declare some variables to make the code clearer

$studentRecord = $data5;
$lineBreak = "\n";
$comma = ", ";


$myArray = explode(',', $_POST["wrongOnes"]);

if($fileName = fopen($studentRecord,"a")) //if there is no file error
{    
    #write the first line in the open doc 
    $string0 = "Test taken: ".$data0;
    $string1 = "Name: ".$data1;
    $string2 = "Score: ".$data2;
    $string3 = "Questions asked: ".$data3;
    $string4 = "Incorrect answers: ".$data4;

    fputs($fileName,$string0.$comma.$lineBreak); 
    fputs($fileName,$string1.$comma.$lineBreak); 
    fputs($fileName,$string2.$comma.$lineBreak);
    fputs($fileName,$string3.$comma.$lineBreak);
    fputs($fileName,$string4.$comma.$lineBreak);
    fputs($fileName,$lineBreak);
    
    #Close the file 
    fclose($fileName); 
}

// variables are sent to this PHP page through
// the POST method.  $_POST is a global associative array
// of variables passed through this method.  From that, we
// can get the values sent to this page from Flash and
// assign them to appropriate variables which can be used
// in the PHP mail() function.


// header information not including sendTo and Subject
// these all go in one variable.  First, include From:
$headers = "From: " . $_POST["realname"]. "<" . ***@***.***".">\r\n";
// next include a replyto
$headers .= "Reply-To: " . ***@***.***". "\r\n";
// often email servers won't allow emails to be sent to
// domains other than their own.  The return path here will
// often lift that restriction so, for instance, you could send
// email to a hotmail account. (hosting provider settings may vary)
// technically bounced email is supposed to go to the return-path email
$headers .= "Return-path: " . $_POST["email"]. "\r\n";;



// now we can add the content of the message to a body variable

$message = $_POST["realname"]." has taken the ".$data0." and attempted ".$_POST["asked"]. " questions";

$message .= " in ".$_POST["timeTakenMin"]." minutes and ".$_POST["timeTakenSec"]." seconds";

$message .= " and scored a total of ".$_POST["marks"]."\r\n\r\n";

$message .= "The results were sent to ".$data6."\r\n\r\n";

$message .= " The following questions were answered incorrectly: ";

foreach($myArray as $my_Array){
    $message .= "\r\n".$my_Array;  
}

$message .= "\r\n\r\n"."The final percentage total = ".$data7."%";

// once the variables have been defined, they can be included
// in the mail function call which will send you an email
mail($sendTo, $subject, $message, $headers);

?>

********************************************************


There are about 20 tests available on www.ibchem.com/ 
Login: guest
Password: guest

Cheers,

Graeme

**Comment by flashuser2** [comment]
Hi,
Got it working. :) thanks a lot.
You were right, n0samu. It wasn’t a flash or a ruffle-problem. It was the php-file. It used to work (before 2020), but testing it in the Flash-enabled browser showed, that something must have changed. After trying a bit of everything, I swopped §sender and §receiver in the php-file and voila, a mail was sent ‘from myschool to the user’.  Even though I wanted it to send a mail the opposite way, - it showed me where the problem was. And after some further “experiments” with the php-files, I got it working the other way around too. 
So now:   - the students can send mails to myschool, and new passwords can be sent to forgetful students, - just like they used to. This was one of the last elements, that had to work in ruffle before my site is up and running again.

Also thanks to you, Graeme.  I have always tried to keep the php-files on my site very simple, as it is really not my homeground, but the one you sent is easy to understand and really useful  -  http://www.ibchem.com/ - very impressive :)  -
